itou/www/job_seekers_views/views.py

```diff
@@ -1244,3 +1245,39 @@ def get_context_data(self, **kwargs):
                 "apply:check_prev_applications_for_hire", kwargs={"session_uuid": self.apply_session.name}
             ),
         }
+
+
+@check_user(lambda user: user.is_job_seeker or user.is_employer or user.is_prescriber)
+def nir_modification_request(request, public_id=None, template_name="job_seekers_views/nir_modification_request.html"):
+    if (public_id and request.user.is_job_seeker) or (not public_id and not request.user.is_job_seeker):
+        raise PermissionDenied
+    if public_id:
+        job_seeker = get_object_or_404(User, public_id=public_id, kind=UserKind.JOB_SEEKER)
+        if not can_view_personal_information(request, job_seeker):
+            raise PermissionDenied
+    else:
+        job_seeker = request.user
+
+    job_application = JobApplication.objects.filter(
+        job_seeker=job_seeker, pk=request.GET.get("jobapplication")
+    ).first()
+    back_url = get_safe_url(request, "back_url", reverse("dashboard:index"))
+    form = NirModificationRequestForm(
+        job_seeker=job_seeker, requestor=request.user, job_application=job_application, data=request.POST or None
+    )
+    is_request_from_proxy = public_id is not None
```

Comment:
Il me semble plus clair de se baser sur le demandeur que sur le passage d’un argument ou non.
Par exemple, la clause du garde `(public_id and request.user.is_job_seeker)` pourrait être modifiée en `(request.user.is_job_seeker and request.user.public_id == public_id)`, ce qui éviterait une 403 pour un cas plus ou moins autorisé. Si ce changement était effectué, `is_request_from_proxy` deviendrait faux.
```suggestion
    is_request_from_proxy = self.request.user != job_seeker
```